### PR TITLE
IW-1632 | linting + notifications for mobile

### PR DESCRIPTION
**Pull request by xkxd**
That's part #2 of notifications in nav.

I've also fixed most (3 that are left are related to sth I'll work on) of linting issues.

https://wikia-inc.atlassian.net/browse/IW-1632

@Wikia/iwing

### CODE REVIEW

File: source/components/GlobalNavigation/components/GlobalNavigationLink/GlobalNavigationLinkLogOut.js [hunk 1]
```diff
@@ -1,31 +1,34 @@
...
-                    >
-                        {t(link.title.key)}
-                    </button>
-                </form>
-            </div>
-        );
-    }
-}
+const GlobalNavigationLinkLogOut = ({ link, t, redirectUrl, className }) => (
+    <div id={link.title.key} className={className || ''}>
```
Comment (mszabo-wikia) [inline source/components/GlobalNavigation/components/GlobalNavigationLink/GlobalNavigationLinkLogOut.js]: should `className` be an optional property?

File: source/components/GlobalNavigation/components/GlobalNavigationNotifications/MarkAllButton.js [hunk 2]
```diff
@@ -20,9 +21,9 @@ const MarkAllButton = ({ track, header }) => {
             {
                 header && <p className="wds-notifications__header">{header}</p>
             }
-            <a onClick={onClick} className="wds-notifications__mark-all-as-read">
+            <button onClick={onClick} className="wds-notifications__mark-all-as-read" role="presentation">
```
Comment (mszabo-wikia) [inline source/components/GlobalNavigation/components/GlobalNavigationNotifications/MarkAllButton.js]: we should remove this eslint rule as I don't think we care about accessibility :/

Comment (xkxd) [reply source/components/GlobalNavigation/components/GlobalNavigationNotifications/MarkAllButton.js]: i've already disabled a few, but yes - seems like we need to get rid of a few others as well

File: source/components/GlobalNavigation/components/GlobalNavigationLink/GlobalNavigationLinkLogOut.js (hunk 1 above)
Comment (xkxd) [reply source/components/GlobalNavigation/components/GlobalNavigationLink/GlobalNavigationLinkLogOut.js]: aye!

**Pull request merged**